packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,241 @@
+   * Labels for landmarks with the same role must also be unique.
+   * 
+   * See https://www.w3.org/TR/wai-aria-practices/examples/landmarks/navigation.html.
+   */
+  private checkLabels(role: AriaLandmarkRole) {
+    let landmarksWithRole = this.getLandmarksByRole(role);
+    if (landmarksWithRole.size > 1) {
+      if ([...landmarksWithRole].some(landmark => !landmark.label)) {
+        console.warn(`Page contains more than one landmark with the '${role}' role. If two or more landmarks on a page share the same role, all must be labeled with an aria-label or aria-labelledby attribute.`);
+      } else {
+        let labels = [...landmarksWithRole].map(landmark => landmark.label);
+        let duplicateLabels = labels.filter((item, index) => labels.indexOf(item) !== index);
+  
+        duplicateLabels.forEach((label) => {
+          console.warn(`Page contains more than one landmark with the '${role}' role and '${label}' label. If two or more landmarks on a page share the same role, they must have unique labels.`);
+        });
+      }
+    }
+  }
+
+  /**
+   * Get the landmark that is the closest parent in the DOM.
+   * Returns undefined if no parent is a landmark.
+   */
+  private closestLandmark(element: HTMLElement) {
+    let landmarkMap = new Map(this.landmarks.map(l => [l.ref.current, l]));
+    let currentElement = element;
+    while (!landmarkMap.has(currentElement) && currentElement !== document.body) {
+      currentElement = currentElement.parentElement;
+    }
+    return landmarkMap.get(currentElement);
+  }
+
+  /**
+   * Gets the next landmark, in DOM focus order, or previous if backwards is specified.
+   * If nested, next should be the child landmark. 
+   * If last landmark, next should be the first landmark.
+   * If not inside a landmark, will return first landmark.
+   * Returns undefined if there are no landmarks.
+   */
+  public getNextLandmark(element: HTMLElement, {backward}: {backward?: boolean }) {
+    if (this.landmarks.length === 0) {
+      return undefined;
+    }
+    
+    let currentLandmark = this.closestLandmark(element);
+    let nextLandmarkIndex = backward ? -1 : 0;
+    if (currentLandmark) {
+      nextLandmarkIndex = this.landmarks.findIndex(landmark => landmark === currentLandmark) + (backward ? -1 : 1);
+    }
+
+    // Wrap if necessary
+    if (nextLandmarkIndex < 0) {
+      nextLandmarkIndex = this.landmarks.length - 1;
+    } else if (nextLandmarkIndex >= this.landmarks.length) {
+      nextLandmarkIndex = 0;
+    }
+
+    return this.landmarks[nextLandmarkIndex];
+  }
+
+  /**
+   * Look at next landmark. If an element was previously focused inside, restore focus there.
+   * If not, focus the first focusable element inside the lanemark.
+   * If no focusable elements inside, go to the next landmark.
+   * If no landmarks at all, or none with focusable elements, don't move focus.
+   */
+  public f6Handler(e: KeyboardEvent) {
```

Comment:
Is `Alt` supposed to be handled?

Reply:
what would alt do?

Reply:
From https://github.com/adobe/react-spectrum/issues/2353, `Alt+F6` should go to the main landmark

Reply:
Great catch, I'll add support for that

Reply:
Added support for `Alt+F6`